baseline/sweep

```diff
@@ -42,41 +54,92 @@ train() {
 
 ensemble() {
     for LEVEL in low medium high; do
-        # TODO(kbg): Add test ensembles when test data becomes available.
-        for DEV in "${DATA}/${LEVEL}/"*"_dev.tsv"; do
-            LG="$(basename ${DEV} _dev.tsv)"
-            python trans/ensembling.py \
-                --gold="${DATA}/${LEVEL}/${LG}_dev.tsv" \
-                --systems "${OUTPUT}/${LEVEL}/${LG}/"*"/dev_beam${BEAM_WIDTH}.predictions" \
-                --output="${OUTPUT}/${LEVEL}/${LG}/ensemble"
+        for TRAIN in "${DATA}/${LEVEL}/"*"_train.tsv"; do
+            LG="$(basename ${TRAIN} _train.tsv)"
+            if [ ${LANGS} ] && ! [[ " ${LANGS} " =~ " ${LG} " ]]
+                # If we've passed target languages, we wish to ignore all non-targets.
+                then
+                    continue
+            fi
+            OUTPUTPATH="${OUTPUT}/${LEVEL}/${LG}/ensemble"
+            if  [ -d ${OUTPUTPATH} ] && [ "$(ls -A ${OUTPUTPATH})" ] && ! [ ${LANGS} ]
+                # We've already trained on this data up to this ensemble. So no need to do again unless we're looking for a specific language.
+                then
+                    continue
+            fi
+            for SPLIT in dev test; do
+                    python trans/ensembling.py \
+                        --gold="${DATA}/${LEVEL}/${LG}_${SPLIT}.tsv" \
+                        --systems "${OUTPUT}/${LEVEL}/${LG}/"*"/${SPLIT}_beam${BEAM_WIDTH}.predictions" \
+                        --output="${OUTPUTPATH}"
+	        done
         done
     done
 }
 
 evaluate() {
+    # Creates two-column TSV with gold and hypothesis data.
     for LEVEL in low medium high; do
-        # TODO(kbg): Revert to test path when test data becomes available.
-        for DEV in "${DATA}/${LEVEL}/"*"_dev.tsv"; do
-            LG="$(basename ${DEV} _dev.tsv)"
-            paste \
-                "${DEV}" \
-                "${OUTPUT}/${LEVEL}/${LG}/ensemble/dev_${MAX_ENSEMBLE_SIZE}ensemble.predictions" \
-            | cut -f2,4 \
-            > "${OUTPUT}/${LEVEL}/${LG}/ensemble/dev_${MAX_ENSEMBLE_SIZE}ensemble.tsv"
+        for TRAIN in "${DATA}/${LEVEL}/"*"_train.tsv"; do
+            LG="$(basename ${TRAIN} _train.tsv)"
+            for SPLIT in dev test; do
+                paste \
+                "${TRAIN//_train.tsv/_${SPLIT}.tsv}" \
+                    "${OUTPUT}/${LEVEL}/${LG}/ensemble/${SPLIT}_${MAX_ENSEMBLE_SIZE}ensemble.predictions" \
+                    | cut -f2,4 \
+                > "${OUTPUT}/${LEVEL}/${LG}/ensemble/${SPLIT}_${MAX_ENSEMBLE_SIZE}ensemble.tsv"
+            done
+        done
+    done
+    
+    # Calls evaluation script.
+    for SPLIT in dev test; do
+        echo "Split: ${SPLIT}"
+        for LEVEL in low medium high; do
+            echo "Level: ${LEVEL}"
+                python ${EVALUATION}/./evaluate_all.py \
+                    "${OUTPUT}/"*"/"*"/ensemble/${SPLIT}_${MAX_ENSEMBLE_SIZE}ensemble.tsv"
```

Comment:
I think the old logic was right here and this breaks it?

Reply:
Yep, you're right. I was fiddling around with eval and forgot to revert that one. I do think on a later edit the directory structure should be changed so it only evaluates existing beam predictions instead of assuming which ones exist.

Reply:
Fine with me for future work.